### ISSUE TITLE
AttributeError: 'NodeContext' object has no attribute 'runtime_properties' in /neutron_plugin/floatingip.py

### ISSUE THREAD

**Issue by SAkagiI** [cloudify-cosmo/cloudify-openstack-plugin#226]
Executing install workflow was failed in /neutron_plugin/floatingip.py.

## My Environments
- CentOS 7.4
  - Cloudify Manager 4.3
- cloudify-openstack-plugin 2.7.1

## Error Message
```
[__main__] ERROR: Task neutron_plugin.floatingip.create[6359de12-4d2b-44aa-ad97-e7a09f6f6538] raised:

Traceback (most recent call last):

  File "/opt/mgmtworker/env/lib/python2.7/site-packages/cloudify/dispatch.py", line 641, in main

    payload = handler.handle()

  File "/opt/mgmtworker/env/lib/python2.7/site-packages/cloudify/dispatch.py", line 397, in handle

    result = self.func(*self.args, **kwargs)

  File "/opt/mgmtworker/env/plugins/default_tenant/cloudify-openstack-plugin-2.7.1/lib/python2.7/site-packages/openstack_plugin_common/__init__.py", line 850, in wrapper

    return f(*args, **kw)

  File "/opt/mgmtworker/env/plugins/default_tenant/cloudify-openstack-plugin-2.7.1/lib/python2.7/site-packages/neutron_plugin/floatingip.py", line 67, in create

    network_from_rel = connected_network.runtime_properties[

AttributeError: 'NodeContext' object has no attribute 'runtime_properties'

```

## Blueprints
```
  floating_ip:
    type: cloudify.openstack.nodes.FloatingIP
    properties:
      openstack_config: { get_input: openstack_configuration }
      floatingip:
        floating_network_name: { get_input: external_network_name }
    relationships:
      - type: cloudify.relationships.depends_on
        target: external_network_subnet
      - type: cloudify.relationships.depends_on
        target: external_network
```

## Source Code

Error Message means connected_network is 'NodeContext' object wchich has no runtime_properties.

The definition of connected_network is get_single_connected_node_by_openstack_type() which return 'NodeContext' object.

- floatingip.py
  https://github.com/cloudify-cosmo/cloudify-openstack-plugin/blob/master/neutron_plugin/floatingip.py#L67
```
#62    connected_network = \
#63        get_single_connected_node_by_openstack_type(
#64         ctx, NETWORK_OPENSTACK_TYPE, True)
#65
#66    if connected_network:
#67        network_from_rel = connected_network.runtime_properties[
#68                             OPENSTACK_ID_PROPERTY]
```

- get_single_connected_node_by_openstack_type()
  https://github.com/cloudify-cosmo/cloudify-openstack-plugin/blob/master/openstack_plugin_common/__init__.py#L263
```
#263   def get_single_connected_node_by_openstack_type(
#264       ctx, type_name, if_exists=False):
#265    nodes = get_connected_nodes_by_openstack_type(ctx, type_name)
#266   check = len(nodes) > 1 if if_exists else len(nodes) != 1
#267    if check:
#268       raise NonRecoverableError(
#269            'Expected {0} one {1} node. got {2}'.format(
#270                'at most' if if_exists else 'exactly', type_name, len(nodes)))
#271    return nodes[0] if nodes else None
```

## Test Code

get_single_connected_node_by_openstack_type was changed MockNodeInstanceContext()  in Test Code.
I think connected_network was expected to be 'NodeInstanceContext' object.

- mock.patch( 'get_single_connected_node_by_openstack_type', MockNodeInstanceContext() )
  https://github.com/cloudify-cosmo/cloudify-openstack-plugin/blob/master/neutron_plugin/tests/test_floating_ip.py#L35
  https://github.com/cloudify-cosmo/cloudify-openstack-plugin/blob/master/neutron_plugin/tests/test_floating_ip.py#L49

```
#18    def test_network_rel(self, cfy_local, *_):
#19       def _mock_rel(*_):
#20            return MockNodeInstanceContext(runtime_properties={
#21               OPENSTACK_ID_PROPERTY: 'my-id'
#22            })
#23
#24        def _mock_create(_, fip):
#25            self.assertEqual(fip[FLOATINGIP_OPENSTACK_TYPE][
#26                                 'floating_network_id'], 'my-id')
#27            return {FLOATINGIP_OPENSTACK_TYPE: {
#28                'id': '1234',
#29                'floating_ip_address': '1.2.3.4'
#30            }}
#31
#32       with mock.patch('neutronclient.v2_0.client.Client.create_floatingip',
#33                        new=_mock_create):
#34            with mock.patch(
#35                    'neutron_plugin.floatingip.get_single_connected_node_by_'
#36                    'openstack_type', new=_mock_rel):
#37                            cfy_local.execute('install')
```

**Comment by SAkagiI** [comment]
I checked the commit history of floatingip.py. 
- https://github.com/cloudify-cosmo/cloudify-openstack-plugin/commits/master/neutron_plugin

「delete get_single_connected_node_by_openstack_type() and use get_openstack_id_of_single_connected_node_by_openstack_type()」in "Some Floating IP Changes" commit.
- https://github.com/cloudify-cosmo/cloudify-openstack-plugin/commit/19270030957421dead10d2f5b51f58d101d1f218#diff-b676dd22b767e05ddb79a8bc9ecc1c9a

I suppose this issue was improved by changing floatingip.py, line 67. 
However, I will not close this issue until verification is completed.

**Comment by EarthmanT** [comment]
Issue has been fixed in version 2.9.0

**Comment by SAkagiI** [comment]
OK, Thank you for fixing this isuue.
